sshfs.c

```diff
@@ -849,11 +879,35 @@ static void ssh_add_arg(const char *arg)
 #ifdef SSH_NODELAY_WORKAROUND
 static int do_ssh_nodelay_workaround(void)
 {
+#ifdef __APPLE__
+	char *oldpreload = getenv("DYLD_INSERT_LIBRARIES");
+#else
 	char *oldpreload = getenv("LD_PRELOAD");
+#endif
 	char *newpreload;
 	char sopath[PATH_MAX];
 	int res;
 
+#ifdef __APPLE__
+	char *sshfs_program_path_base = NULL;
+	if (!sshfs_program_path[0]) {
+		goto nobundle;
+	}
+	sshfs_program_path_base = dirname(sshfs_program_path);
+	if (!sshfs_program_path_base) {
+		goto nobundle;
+	}
+	snprintf(sopath, sizeof(sopath), "%s/%s", sshfs_program_path_base,
+             SSHNODELAY_SO);
+	res = access(sopath, R_OK);
+	if (res == -1) {
+		goto nobundle;
+	}
+	goto pathok;
+
+nobundle:
+#endif /* __APPLE__ */
```

Comment:
I don't know for sure, but "bundles" in OS X are a way of shipping multiple files as a single entity. The UI displays such entities as regular files (and they are handled as a unit), but underneath they are actually directories with a specific file layout.

I suppose that either MacFUSE or OSXFUSE are distributing sshfs as a bundle and that within the bundle is the `sshnodelay.so` file. This trick allows for the library to be found within the bundle instead of in the system path.